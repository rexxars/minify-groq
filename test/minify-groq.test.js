const minify = require('../src/minify-groq')

test('handles empty strings', () => {
  const input = ``
  const expected = ``
  expect(minify(input)).toEqual(expected)
})

test('does not strip whitespace from double string', () => {
  const input = `"foo bar\tbaz\nzing  multi space"`
  const expected = input
  expect(minify(input)).toEqual(expected)
})

test('does not strip whitespace from single string', () => {
  const input = `'foo bar\tbaz\nzing  multi space'`
  const expected = input
  expect(minify(input)).toEqual(expected)
})

test('handles escaped double strings', () => {
  const input = `"it's \\"escaped\\", ok?"`
  const expected = input
  expect(minify(input)).toEqual(expected)
})

test('handles escaped single strings', () => {
  const input = `'it\\'s "escaped", ok?'`
  const expected = input
  expect(minify(input)).toEqual(expected)
})

test('handles full query', () => {
  const input = `
    {
      "basic test": 'works',
      "test \\"escaped\\"": '\\'works\\'',
      "draft": *[_id in path("drafts.**")] [ 0 ],

      'it\\'s\n\t"aliased"': * [ _type ==  "author"   ] {
        ... ,
        name,
        "cat \\"C:\\test\\\\\\"spaced file.txt\\\\\\"": awards[ @ == "Noe  \\"optimistisk\\"" ]
      }  |  order( _createdAt desc ) [ 0 ]  [ 0  ...  2 ]  ,

      "static": 'value   \n   with spacing'
    }
  `

  expect(minify(input)).toMatchSnapshot()
})

test('handles multiple pipes', () => {
  const input = `
    * [ _type == "author" && birthYear > 1950] {
      name,
      birthYear,
      "books": books[]->{
        title
      }
    } | order(birthYear asc) | [ 0 ... 50 ]
  `

  expect(minify(input)).toMatchSnapshot()
})

test('handles selects, newlined pipes', () => {
  const input = `
  * [ _type == "author" && birthYear > 1950]
    | order(
      birthYear
      asc
    )
    | [ 0 ... 50 ]
    | {
      name,
      birthYear,

      "popularity": select(
        popularity > 20 => "high",
        popularity > 10 => "medium",
        popularity <= 10 => "low"
      ),

      "works": works[]->{
        _type == "book" => {
          title
        },
        _type == "manuscript" => {
          title,
          kind
        }
      }
    }
  `

  expect(minify(input)).toMatchSnapshot()
})

test('can be used as template literal', () => {
  expect(minify`{  "foo": "\tbar" } `).toEqual(`{ "foo": "\tbar" }`)

  const type = 'book'
  const year = 1950
  const categoryIds = [13, 15]
  expect(
    minify`*[
      _type == ${type} &&
      releaseYear == ${year} &&
      categoryId in ${categoryIds}
    ] {
      title,
      releaseYear,
    }`
  ).toMatchSnapshot()
})
