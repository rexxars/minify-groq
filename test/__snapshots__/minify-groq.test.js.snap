// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can be used as template literal 1`] = `
"*[
_type == \\"book\\" &&
releaseYear == 1950 &&
categoryId in [13,15]
] {
title,
releaseYear,
}"
`;

exports[`handles full query 1`] = `
"{
\\"basic test\\": 'works',
\\"test \\\\\\"escaped\\\\\\"\\": '\\\\'works\\\\'',
\\"draft\\": *[_id in path (\\"drafts.**\\")] [ 0 ],
'it\\\\'s
	\\"aliased\\"': * [ _type == \\"author\\" ] {
... ,
name,
\\"cat \\\\\\"C:\\\\test\\\\\\\\\\\\\\"spaced file.txt\\\\\\\\\\\\\\"\\": awards[ @ == \\"Noe  \\\\\\"optimistisk\\\\\\"\\" ]
} | order ( _createdAt desc ) [ 0 ] [ 0 ... 2 ] ,
\\"static\\": 'value   
   with spacing'
}"
`;

exports[`handles multiple pipes 1`] = `
"* [ _type == \\"author\\" && birthYear > 1950] {
name,
birthYear,
\\"books\\": books[]->{
title
}
} | order (birthYear asc) | [ 0 ... 50 ]"
`;

exports[`handles selects, newlined pipes 1`] = `
"* [ _type == \\"author\\" && birthYear > 1950]
| order (
birthYear
asc
)
| [ 0 ... 50 ]
| {
name,
birthYear,
\\"popularity\\": select(
popularity > 20 => \\"high\\",
popularity > 10 => \\"medium\\",
popularity <= 10 => \\"low\\",
),
\\"books\\": favoriteBooks[]->{
_type == \\"book\\" => {
title
},
_type == \\"manuscript\\" => {
title,
kind
}
}
}"
`;
